### PR TITLE
Support for OPLxLPT music cards

### LINKED ISSUE #71: opl3lpt support how?
I saw there're some functions referring to opl3lpt, and I would love to use my device.

How I could activate them? (even recompiling, if required to)

### PR DESCRIPTION
**Pull request by jsmolina**
Fixes https://github.com/viti95/FastDoom/issues/71

### CODE REVIEW

File: FASTDOOM/ns_sbmus.c [hunk 1]
```diff
@@ -226,6 +195,46 @@ void AL_SendOutputToPort_OPL3LPT(int port, int reg, int data)
...
+void AL_SendOutputToPort(
+    int port,
+    int reg,
+    int data)
+
+{
+   int delay;
+
+   // use OPL3 func for this. TODO use soundcard enum
+   if(ADLIB_PORT == 0x378) {
```
Comment (jsmolina) [inline FASTDOOM/ns_sbmus.c:214]: I guess we'll have to improve this.

File: FASTDOOM/ns_sbmus.c [hunk 2]
```diff
@@ -1145,6 +1154,11 @@ int AL_DetectFM(
    int status2;
    int i;
 
+   // no detection for OPL2LPT or OPL3LPT
+   if (ADLIB_PORT == 0x378) {
```
Comment (jsmolina) [inline FASTDOOM/ns_sbmus.c:1158]: this one as well

**Comment by viti95** [comment]
Can you check if the OPL3LPT device works with the last commit?

**Comment by jsmolina** [comment]
> Can you check if the OPL3LPT device works with the last commit?

looks good! Let me configure the fdsetup compiler and I'll test both.

**Comment by jsmolina** [comment]
@viti95 just compiled. 

fdsetup works fine, but fastdoom does something strange:

- If I select Sound Blaster as MIDI and FX, everything works as expected: so my solo1 outputs FX and Music.
- If I select OPL3LPT and LPT1, then something weird happens: the game hangs and the SOLO1 (yes, the 'SoundBlaster', not the OPL3LPT) does a beep loop.

File: FASTDOOM/ns_music.c [hunk 3]
```diff
@@ -34,10 +34,7 @@ int MUSIC_InitGUS(midifuncs *Funcs);
    Selects which sound device to use.
 ---------------------------------------------------------------------*/
 
-int MUSIC_Init(
-    int SoundCard,
-    int Address)
-
+int MUSIC_Init(int SoundCard, int Address)
```
Comment (jsmolina) [inline FASTDOOM/ns_music.c:37]: I was following the code and I don't see where this address is passed through, 
maybe this is the issue?

**Comment by viti95** [comment]
Does it work if you config only the music device (no sound FX)?

**Comment by jsmolina** [comment]
> Does it work if you config only the music device (no sound FX)?

in such case, there's no hang but also no audio.

I tried OPL3LPT test program and it does output sound. Same with my 'always olp3' test.

File: FASTDOOM/ns_music.c (hunk 3 above)
Comment (viti95) [reply FASTDOOM/ns_music.c:37]: Yep, I think that was the issue. I've uploaded another commit to fix it. Can you try again?

**Comment by jsmolina** [comment]
last commits fixed the issue!

thanks @viti95

**Pull request merged**